doc/script_commands.txt

```diff
@@ -2612,6 +2612,79 @@ deleted.
 	// This would delete all elements of the array
 	deletearray(.@array);
 
+---------------------------------------
+//=====================================
+1.1 - Message formatting commands (tags)
+//=====================================
+---------------------------------------
+
+*mesnavigation("<label>", "<map>"{, <x>{, <y>{, <show_window>{, <mode>{, <services_flag>{, <monster_id>}}}}}});
+
+Generates and returns a <NAVI> tag with the given parameters. A <NAVI> tag when written to certain windows,
+such as the one from "mes" command, will create a highlighted and clickable text, which when clicked,
+starts the navigation system for the specified values.
+
+It is important to note that for most "<modes>" values, it is expected that your client has proper navigation
+files which includes the target NPC/monster, or they will fail.
+
+If the current client doesn't support NAVI tags, returns a fall back text which generally looks like:
+	<label> (<map_name> <x>, <y>)
+
+Parameters:
+- label: (string, required)
+	The visible and clickable text. (example: "Hat maker")
+
+- map: (string, required)
+	Target map name. (example: "prontera")
+	Note: when service_flag is NAV_WINDOW_SEARCH (-222) this becomes the search string.
+
+- x: (int, optional, defaults to 0)
+	Target map X position
+	Note: does nothing when used with modes NAV_MODE_MAP or NAV_MODE_MOB
+
+- y: (int, optional, defaults to 0)
+	Target map Y position
+	Note: does nothing when used with modes NAV_MODE_MAP or NAV_MODE_MOB
+
+- show_window: (bool, optional, defaults to false)
+	Whether to open the navigation UI (true) or not (false) when clicking the label.
+
+- mode: (int, optional, defaults to NAV_MODE_ALL (0))
+	The kind of navigation being performed. This mainly affects the messages shown by the client when you reach the map.
+	- NAV_MODE_ALL (0): "You have reached your destination"
+	- NAV_MODE_MAP (1): "You have reached the target map"
+	- NAV_MODE_NPC (2): "You have reached the map the NPC is at" / "You have reached the target NPC"
```

Comment:
I thought about making a `mesmapnavi("mapname"{, "label"})` and `mesnpcnavi("NPC Name"{, "label"})` which could check whether map or npc exists. But I don't remember seeing those being used in official scripts... so I am not sure if it is worth it.